ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -1890,3 +1892,111 @@ def test028_update_test_unit_with_delete_in_form(self):
+        row = self.orders_page.get_random_order()
+        order_page_url = self.base_selenium.get_url()
+        self.base_selenium.click(element='order:suborder_table')
+
+        initial_article = self.order_page.get_article()
+        initial_test_plan = self.order_page.get_test_plan()
+        initial_test_unit = self.order_page.get_test_unit()
+
+
+        # Update Article
+        self.order_page.set_article(
+            article='r')
+        new_article = self.order_page.get_article()
+
+        # Update Test Plan
+        if self.order_page.get_test_plan():
+            self.order_page.clear_test_plan()
+            self.order_page.confirm_popup(force=True)
+
+        self.order_page.set_test_plan(test_plan='r')
+        new_test_plan = self.order_page.get_test_plan()
+
+        # Update Test Unit
+        if self.order_page.get_test_unit():
+            self.order_page.clear_test_unit()
+            self.order_page.confirm_popup(force=True)
+
+        self.order_page.set_test_unit(test_unit='r')
+        new_test_unit = self.order_page.get_test_unit()
+
+
+        if 'save_btn' == save:
+            self.order_page.save(save_btn='order:save_btn')
+            self.base_selenium.LOGGER.info('Refresh to make sure that data are saved correctly')
+            self.base_selenium.refresh()
+            self.base_selenium.click(element='order:suborder_table')
+            current_article = self.order_page.get_article()
+            current_test_plan = self.order_page.get_test_plan()
+            current_test_unit = self.order_page.get_test_unit()
+        else:
+            self.order_page.cancel(force=True)
+            self.base_selenium.get(url=order_page_url)
+            self.order_page.sleep_tiny()
+            self.base_selenium.click(element='order:suborder_table')
+            current_article = self.order_page.get_article()
+            current_test_plan = self.order_page.get_test_plan()
+            current_test_unit = self.order_page.get_test_unit()
+            
+
+        if 'save_btn' == save:
+            self.base_selenium.LOGGER.info(
+                ' + Assert {} (current_article) == {} (new_article)'.format(current_article,
+                                                                            new_article))
+            self.assertEqual(new_article, current_article)
+
+            self.base_selenium.LOGGER.info(
+                ' + Assert {} (current_test_plan) == {} (new_test_plan)'.format(current_test_plan,
+                                                                            new_test_plan))
+            self.assertEqual(new_test_plan, current_test_plan)
+
+
+            self.base_selenium.LOGGER.info(
+                ' + Assert {} (current_test_unit) == {} (new_test_unit)'.format(current_test_unit,
+                                                                            new_test_unit))
+            self.assertEqual(new_test_unit, current_test_unit)
```

Comment:
we should to assert that make sure that the test unit didn't change we have same value